stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2730,15 +3428,34 @@ impl NakamotoChainState {
             None
         };
 
-        // TODO: only need to do this if this is a tenure-start block
         let (stacking_burn_ops, transfer_burn_ops, delegate_burn_ops, vote_for_agg_key_ops) =
-            StacksChainState::get_stacking_and_transfer_and_delegate_burn_ops(
-                chainstate_tx,
-                &parent_index_hash,
-                sortition_dbconn.sqlite_conn(),
-                &burn_header_hash,
-                burn_header_height.into(),
-            )?;
+            if new_tenure {
```

Comment:
Not convinced this is true -- tenure extends which update the burnchain view should process the new burn ops, right?

Reply:
Yes -- this needs to happen if we have a tenure event.  I.e. the block we're setting up contains a well-formed and valid `TenureChange` transaction

Reply:
So, this should be `if new_tenure || tenure_extend {`